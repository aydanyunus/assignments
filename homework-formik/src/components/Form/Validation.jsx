import * as yup from "yup";

export const SignupSchema = yup.object().shape({
  firstName: yup
    .string()
    .min(2, "Too Short!")
    .max(50, "Too Long!")
    .required("First name is required"),
  lastName: yup
    .string()
    .min(2, "Too Short!")
    .max(50, "Too Long!")
    .required("Last name is required"),
  age: yup
    .number()
    .min(0, "Age cannot be negative")
    .required("Age is required"),
  address: yup
    .string()
    .min(4, "Address must be at least 4 characters")
    .required("Address is required"),
  phoneNumber: yup
    .string()
    .min(10, "Please enter a valid 10-digit phone number")
    .required("Please enter a valid phone number"),
});
