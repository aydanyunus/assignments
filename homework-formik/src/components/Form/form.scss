input[type="text"],
input[type="password"],
input[type="email"],
input[type="number"],
select {
  width: 100%;
  padding: 10px;
  border: none;
  border-radius: 3px;
  font-size: 16px;
}

input:focus {
  outline: none;
}

label {
  font-weight: 400;
  display: block;
  font-size: 16px;
  margin-bottom: 5px;
}

.btn-checkout {
  background-color: #3498db;
  color: #fff;
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  font-size: 16px;
  cursor: pointer;
  margin-top: 15px;
  margin-bottom: 15px;
}

.btn-checkout:hover {
  background-color: #2980b9;
}

.error-message {
  margin-bottom: 15px;
  color: #db1414;
  font-weight: 400;
}

.checkout-title {
  margin-bottom: 20px;
  text-align: center;
}
