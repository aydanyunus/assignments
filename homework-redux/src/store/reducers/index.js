import { combineReducers } from "redux";
import { productsReducer } from './productsReducer';
import { favoritesReducer } from "./favoritesReducer";
import { cartReducer } from "./cartReducer";
import { modalReducer } from "./modalReducer";



export const allReducers = combineReducers({
    products: productsReducer,
    favorites: favoritesReducer,
    cart: cartReducer,
    modal: modalReducer
})
