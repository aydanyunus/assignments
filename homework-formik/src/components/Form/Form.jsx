import React from "react";
import { Formik, Form, Field, ErrorMessage } from "formik";
import { SignupSchema } from "./Validation";
import "./form.scss";
import { useDispatch } from "react-redux";
import { clearCart } from "../../actions/index";

const UserForm = ({ products }) => {
  const dispatch = useDispatch();

  return (
    <div>
      <Formik
        initialValues={{
          firstName: "",
          lastName: "",
          age: "",
          address: "",
          phoneNumber: "",
        }}
        validationSchema={SignupSchema}
        onSubmit={(values) => {
          console.log(values);
          console.log(products);
          dispatch(clearCart());
        }}
      >
        {({ isSubmitting }) => (
          <Form>
            <label htmlFor="firstName">First name</label>
            <Field type="text" name="firstName" id="firstName" />
            <ErrorMessage
              className="error-message"
              name="firstName"
              component="div"
            />

            <label htmlFor="lastName">Last name</label>
            <Field type="text" name="lastName" id="lastName" />
            <ErrorMessage
              className="error-message"
              name="lastName"
              component="div"
            />

            <label htmlFor="age">Age</label>
            <Field type="number" name="age" id="age" />
            <ErrorMessage
              className="error-message"
              name="age"
              component="div"
            />

            <label htmlFor="address">Delivery Address</label>
            <Field type="text" name="address" id="address" />
            <ErrorMessage
              className="error-message"
              name="address"
              component="div"
            />

            <label htmlFor="phoneNumber">Phone Number</label>
            <Field type="text" name="phoneNumber" id="phoneNumber" />
            <ErrorMessage
              className="error-message"
              name="phoneNumber"
              component="div"
            />

            <button
              className="btn-checkout"
              type="submit"
              disabled={isSubmitting}
            >
              Checkout
            </button>
          </Form>
        )}
      </Formik>
    </div>
  );
};

export default UserForm;
