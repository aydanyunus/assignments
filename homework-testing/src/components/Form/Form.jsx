import React from "react";
import { Formik, Form, Field, ErrorMessage } from "formik";
import { SignupSchema } from "./Validation";
import "./form.scss";
import { useDispatch, useSelector } from "react-redux";
import { clearCart } from "../../store/actions/cartActions";

const UserForm = () => {
  const dispatch = useDispatch();
  const cart = useSelector((state) => state.cart.cart);


  return (
    <div>
      <Formik
        initialValues={{
          firstName: "",
          lastName: "",
          age: "",
          address: "",
          phoneNumber: "",
        }}
        validationSchema={SignupSchema}
        onSubmit={(values) => {
          console.log(values);
          console.log(cart);
          dispatch(clearCart());
        }}
      >
        {({ isSubmitting }) => (
          <Form>
            <label htmlFor="firstName">First name</label>
            <Field type="text" name="firstName" id="firstName" />
            <ErrorMessage
              className="error-message"
              name="firstName"
              component="div"
            />

            <label htmlFor="lastName">Last name</label>
            <Field type="text" name="lastName" id="lastName" />
            <ErrorMessage
              className="error-message"
              name="lastName"
              component="div"
            />

            <label htmlFor="age">Age</label>
            <Field type="number" name="age" id="age" />
            <ErrorMessage
              className="error-message"
              name="age"
              component="div"
            />

            <label htmlFor="address">Delivery Address</label>
            <Field type="text" name="address" id="address" />
            <ErrorMessage
              className="error-message"
              name="address"
              component="div"
            />

            <label htmlFor="phoneNumber">Phone Number</label>
            <Field type="text" name="phoneNumber" id="phoneNumber" />
            <ErrorMessage
              className="error-message"
              name="phoneNumber"
              component="div"
            />

            <button
              className="btn-checkout"
              type="submit"
              disabled={isSubmitting}
            >
              Checkout
            </button>
          </Form>
        )}
      </Formik>
    </div>
  );
};

export default UserForm;
