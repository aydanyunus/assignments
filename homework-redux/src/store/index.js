import { createStore } from "redux";
import { allReducers } from '../reducers/index';
import {applyMiddleware} from "redux";
import thunkMiddleware from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension';



const store = createStore(
    allReducers,
    composeWithDevTools(
        applyMiddleware(thunkMiddleware)
    )
)

export default store;