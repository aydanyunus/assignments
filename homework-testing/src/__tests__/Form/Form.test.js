import Form from '../../components/Form/Form';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store/index';
import renderer from 'react-test-renderer';




test('Form component snapshot', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <BrowserRouter>
                    <Form />
                </BrowserRouter>
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});