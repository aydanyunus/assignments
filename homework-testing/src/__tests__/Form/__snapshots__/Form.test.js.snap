// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot 1`] = `
<div>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <label
      htmlFor="firstName"
    >
      First name
    </label>
    <input
      id="firstName"
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="lastName"
    >
      Last name
    </label>
    <input
      id="lastName"
      name="lastName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="age"
    >
      Age
    </label>
    <input
      id="age"
      name="age"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      htmlFor="address"
    >
      Delivery Address
    </label>
    <input
      id="address"
      name="address"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="phoneNumber"
    >
      Phone Number
    </label>
    <input
      id="phoneNumber"
      name="phoneNumber"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="btn-checkout"
      disabled={false}
      type="submit"
    >
      Checkout
    </button>
  </form>
</div>
`;
